lib/codeowners/cli/interactive_fix.rb

```diff
@@ -0,0 +1,92 @@
+# frozen_string_literal: true
+
+require 'forwardable'
+require_relative 'interactive_helpers'
+require_relative './helpers/assign_file_owner'
+require_relative './helpers/add_pattern_into_group'
+require_relative './helpers/suggest_subgroup_for_pattern'
+
+module Codeowners
+  module Cli
+    # Command Line Interface which covers dialogues and logic
+    # that are specific for adding new pattern.
+    #
+    # Process:
+    #
+    # - Program will ask if need to add owner for new file detected or ignore it.
+    #
+    #   EG:
+    #
+    #     File added: "new_file.rb". Add owner to the CODEOWNERS file?
+    #     (y) yes
+    #     (i) ignore
+    #     (q) quit and save
+    #     [y, i, q]
+    #
+    # - if user prompts (y)es, then:
+    #
+    #   - program will point user into 'choose owner' and 'new file pattern generation'
+    #     (via Helpers::AssignFileOwner class)
+    #
+    #   - if new file pattern generated successfully, then program will
+    #     call Helpers::AddPatternIntoGroup class, where it will:
+    #
+    #     - suggest to add new file pattern into one of existing subgroups if there are any owned by selected owner
+    #       (via Helpers::SuggestSubgroupForPattern class)
+    #
+    #     - or will ask to assign pattern to main group (end of the CODEOWNERS file)
+    #
+    # - if user prompts (i)gnore, then
+    #   program will returns from #suggest_add_to_codeowners method.
+    #
+    # - if user prompts (q)uit, then whole program will be interrupted.
+    #
+    # INPUT:
+    #   New detected file path ('test.rb', 'lib/tasks/generate.rake' so on)
+    #
+    # OUTPUT:
+    #   we are not returning specific value here, just executing list of commands.
+    #
+    class InteractiveFix < Base
```

Comment:
Any reason to make `InteractiveFix` inherits `Base`?

Because it has `no_commands` to all the methods, it does not add any extra commands.  We also depend on `cli` and other things that belong to the main command.

I'd vote to make it a module and inherit it dynamically only if `options[:interactive]` is true.